include/zephyr/location.h

```diff
@@ -0,0 +1,340 @@
+/**
+ * @file location.h
+ *
+ * @brief Public location API.
+ *
+ */
+
+/*
+ * Copyright (c) 2022, Trackunit A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_LOCATION_H_
+#define ZEPHYR_INCLUDE_LOCATION_H_
+
+/**
+ * @brief Location Interface
+ * @defgroup location_interface Location Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Location data
+ *
+ * The latitude and longitude are in the unit micro degree. 1 micro
+ * degree corresponds to approx. 11cm along the earths surface.
+ *
+ * The accuracy describes the radius within which the real
+ * position is with a 68.2% certainty, in millimeters.
+ */
+struct location_position {
+	/** Latitudal position in microdegrees (0 to +-180000000) */
+	int32_t latitude;
+	/** Longitudal position in microdegrees (0 to +-180000000) */
+	int32_t longitude;
+	/** The uptime when the position was determined */
+	int64_t uptime;
+	/** Accuracy radius in millimeters */
+	uint32_t accuracy;
```

Comment:
Just wondering what module provides this data? AFAIK we aren't able to get this from typical NMEA sentences. Also, is this `accuracy` supposed to be the same for other metrics like speed/bearing?

Reply:
The accuracy calculation is completely dependent on the technology used to determine the location. For GNSS, it would be based on HDOP or similar. See https://web.archive.org/web/20160310132600/http://edu-observatory.org/gps/gps_accuracy.html for a guide to determining accuracy from this value. Some manufacturers may also provide tables with this data.